### ISSUE TITLE
program hangs because of active ExecutorService

### ISSUE THREAD

**Issue by uwemaurer** [slackapi/java-slack-sdk#396]
### Issue Type

Place an `x` in one of the `[ ]`.

- [x] Bug
- [ ] Enhancement / Feature request
- [ ] Question
- [ ] Documentation

### Description

A simple example program hangs because of an active ExecutorService

---
(only for a bug report)

The issue is reproducible in:
* Module Version:  1.0.0-RC2
* JDK Version:  java 11
* OS Version etc:  ubuntu

The steps to reproduce are:
run this program:
```
public class TestSlack {
    public static void main(String args[]) {
        var a = SlackConfig.DEFAULT;
    }
}

```
The expected result is:
program should end

The actual result is:

program hangs

I think the reason is that `MethodsConfig.DEFAULT_SINGLETON` has a reference to a `MemoryMetricsDatastore` which has a non daemon thread (`Executors.newSingleThreadScheduledExecutor();`)

A fix would be to make this a daemon thread, or not hold on to the reference from the singleton so that it can be finalized
---

### Requirements (place an `x` in each of the `[ ]`)

* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

**Comment by seratch** [comment]
Thank you very much for pointing this out. I've forgotten to do `setDaemon` for those in this library... I will fix it today and will be releasing a new RC version with other planned improvements soon.

**Comment by seratch** [comment]
@uwemaurer I just released version 1.0.0-RC3. Thanks again for pointing out the issue before releasing v1.0.0!

**Comment by uwemaurer** [comment]
@seratch thank you for fixing and releasing a new version!
